protocol/tx/entry.go

```diff
@@ -73,6 +97,10 @@ func writeForHash(w io.Writer, c interface{}) error {
 	case bc.AssetID:
 		_, err := w.Write(v[:])
 		return errors.Wrap(err, "writing bc.AssetID for hash")
+
+	case *EntryRef:
```

Comment:
It would be nice to avoid having extra cases in the hashing logic to accommodate this. How do you feel about storing the pointer fields in a separate parallel structure outside the `body` of each entry?